### ISSUE TITLE
[Offline] expiration field not set & updated in stored manifest when `config.offline.usePersistentLicense` is set to false

### ISSUE THREAD

**Issue by loicraux** [shaka-project/shaka-player#6335]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**If the problem is related to FairPlay, have you read the tutorial?**
N/A

**What version of Shaka Player are you using?**
Latest version 4.7.11

**Can you reproduce the issue with our latest release version?**
Yes

**Can you reproduce the issue with the latest code from `main`?**
Yes

**Are you using the demo app or your own custom app?**
Own custom app

**If custom app, can you reproduce the issue using our demo app?**
N/A (Cannot be tested with demo app, since expiration time is not displayed)

**What browser and OS are you using?**
```
{
  "electronVersion": "28.1.3",
  "nodeVersion": "18.18.2",
  "chromeVersion": "120.0.6099.199",
  "shakaPlayerVersion": "4.7.11",
  "widevineCDMVersion": "4.10.2710.0",
  "operatingSystem": "Linux x64 6.5.0-21-generic"
}
```

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**
N/A

**What are the manifest and license server URIs?**
Not related to a specific manifest and/or license server.

**What configuration are you using?  What is the output of `player.getConfiguration()`?**
`config.offline.usePersistentLicense` is set to `false`.

**What did you do?**

With `offline.usePersistentLicense` set to `false` (meaning we don't persist the licenses), we proceed to download some content using `Storage.store` API and a manifest is created and saved in IndexedDB.

**What did you expect to happen?**

1. The store manifest should have an `expiration` field properly initialized once it has been downloaded (that is to say not equal to Infinity),
2. and when the downloaded content is played (which requires additional licenses requests, since we did not persist any license, because of the setting), if any update on the expiration time happens, these updates should also be persisted in the stored manifest.

**What actually happened?**
The manifest has an `expiration` prop set to `Infinity` (basically it has been initialized to `Infinity` and never updated during the download process) after the download.

Also, when the content is played, the expiration field of the stored manifest remains equal to Infinity.

Note that listeners of the `expirationUpdated(sessionId, expiration)` event are properly called with new, correct expiration times, even when `config.offline.usePersistentLicense` is set to `false`

Basically, I think that this `expiration` field of the stored manifest should be properly set & updated, not matter if `config.offline.usePersistentLicense` is set to `false` or `true`...
Do you agree with this ?

**Comment by joeyparrish** [comment]
> Basically, I think that this `expiration` field of the stored manifest should be properly set & updated, not matter if `config.offline.usePersistentLicense` is set to `false` or `true`...
> Do you agree with this ?

No.  I think I can understand why you would be confused about this, though.

What might not be obvious is that the expiration is for the license.  Content has no expiration.  If there's no license stored, there's no expiration to store.

If you store without a persistent license, then play it later with a temporary license, and that temporary license expires... you can still play it again with another temporary license.  At least as far as EME is concerned.  There's no license state in the browser between those two sessions, so there's no reason the player should think it can't be played.

The license server can always refuse to issue a second temporary license later to the same user, but that's a matter of that license server's policy, and not something the player can know about from EME.

Does this answer your question?
